--- a/include/qemu/timer.h        2022-05-31 14:48:30.687026318 -0500
+++ b/include/qemu/timer.h    2022-05-31 15:45:39.784464685 -0500
@@ -4,6 +4,9 @@
 #include "qemu/bitops.h"
 #include "qemu/notify.h"
 #include "qemu/host-utils.h"
+#ifdef CONFIG_SYSMODE_FUZZER
+#include "sysmode-fuzzer/time-dialation.h"
+#endif

 #define NANOSECONDS_PER_SECOND 1000000000LL

@@ -800,7 +803,9 @@
 /*
  * Low level clock functions
  */
+extern int64_t clock_start;

+#ifndef CONFIG_SYSMODE_FUZZER
 /* get host real time in nanosecond */
 static inline int64_t get_clock_realtime(void)
 {
@@ -810,7 +815,6 @@
     return tv.tv_sec * 1000000000LL + (tv.tv_usec * 1000);
 }

-extern int64_t clock_start;

 /* Warning: don't insert tracepoints into these functions, they are
    also used by simpletrace backend and tracepoints would cause
@@ -842,6 +846,7 @@
     }
 }
 #endif
+#endif

 /*******************************************/
 /* host CPU ticks (if available) */
